### PR TITLE
Fix `updateUser` options

### PR DESCRIPTION
**Pull request by danerwilliams**
## Description

`updateUser` options should require at least one property, not all properties to be passed. 

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: src/users/serializers/update-user-options.serializer.ts [hunk 1]
```diff
@@ -2,7 +2,13 @@ import { SerializedUpdateUserOptions, UpdateUserOptions } from '../interfaces';
...
-});
+): SerializedUpdateUserOptions => {
+  if (Object.keys(options).length === 0) {
+    throw new Error('At least one property must be provided');
+  }
+
+  return {
+    first_name: options.firstName,
+    last_name: options.lastName,
+  } as SerializedUpdateUserOptions;
```
Comment (danerwilliams) [inline src/users/serializers/update-user-options.serializer.ts:13]: I don't like using an `as` here, but it seems like the only way to get this to compile (besides maybe being exceptionally verbose).

Adding a check to throw an error if for some reason there weren't at least 1 property passed I think works just as well here.

Comment (maxdeviant) [reply src/users/serializers/update-user-options.serializer.ts:13]: I think this is a case where we want to use [`satisfies`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#the-satisfies-operator).

The SDK targets TypeScript 5, so we should have access to it.

File: src/users/serializers/update-user-options.serializer.ts [hunk 2]
```diff
@@ -2,7 +2,13 @@ import { SerializedUpdateUserOptions, UpdateUserOptions } from '../interfaces';
 
 export const serializeUpdateUserOptions = (
   options: UpdateUserOptions,
-): SerializedUpdateUserOptions => ({
-  first_name: options.firstName,
-  last_name: options.lastName,
-});
+): SerializedUpdateUserOptions => {
+  if (Object.keys(options).length === 0) {
```
Comment (maxdeviant) [inline src/users/serializers/update-user-options.serializer.ts:6]: nit: More idiomatic to write the length check like this:

```suggestion
  if (!Object.keys(options).length) {
```

**Review by maxdeviant — APPROVED** [review]

File: src/users/serializers/update-user-options.serializer.ts (hunk 1 above)
Comment (danerwilliams) [reply src/users/serializers/update-user-options.serializer.ts]: Wasn't able to get `satisfies` to help with this. 

I thought I had tried this but I can just list out all of the properties in a check though to compile:
```ts
export const serializeUpdateUserOptions = (
  options: UpdateUserOptions,
): SerializedUpdateUserOptions => {
  if (!options.firstName || !options.lastName) {
    throw new Error('At least one of the properties must be provided');
  }

  return {
    first_name: options.firstName,
    last_name: options.lastName,
  };
};
```

kinda annoying but I think it's better than using an `as` with the generic check for empty object

Comment (danerwilliams) [reply src/users/serializers/update-user-options.serializer.ts]: oh wait lol this doesn't make logical sense that's why I didn't do it

Comment (danerwilliams) [reply src/users/serializers/update-user-options.serializer.ts]: need my brain to wake up haha

Comment (danerwilliams) [reply src/users/serializers/update-user-options.serializer.ts:13]: I think we need to just stick with the `as`:

```ts
export const serializeUpdateUserOptions = (
  options: UpdateUserOptions,
): SerializedUpdateUserOptions => {
  if (!Object.keys(options).length) {
    throw new Error('At least one of the properties must be provided');
  }

  return {
    first_name: options.firstName,
    last_name: options.lastName,
  } as SerializedUpdateUserOptions;
};
```

`satisfies` won't narrow down the type down any further in this case

**Pull request merged**